src/github.com/getlantern/flashlight/logging/logging.go

```diff
@@ -179,24 +179,28 @@ func (w logglyErrorWriter) Write(b []byte) (int, error) {
 
 	// extract last 2 (at most) chunks of fullMessage to message, without prefix,
 	// so we can group logs with same reason in Loggly
-	parts := strings.Split(fullMessage, ":")
-	var message string
-	pl := len(parts)
-	switch pl {
-	case 1:
-		message = ""
-	case 2:
-		message = parts[1]
-	default:
-		message = parts[pl-2] + ":" + parts[pl-1]
+	lastColonPos := -1
+	colonsSeen := 0
+	for p := len(fullMessage) - 2; p >= 0; p-- {
+		if fullMessage[p] == ':' {
+			lastChar := fullMessage[p+1]
+			// to prevent colon in "http://" and "x.x.x.x:80" be treated as seperator
+			if !(lastChar == '/' || lastChar >= '0' && lastChar <= '9') {
```

Comment:
Hm, the motive of truncating message is not to miss some item when grouping by it (https://github.com/getlantern/lantern/issues/2287), and we always have `fullMessage`. Grouping by location and by reason can be of different dimension. Anyway, it is insignificant I believe.